### PR TITLE
Fixes #3055: Remove `Key.Unknown`

### LINKED ISSUE #3055: Remove the `KeyCode.Unknown`
> > > I propose to remove the `KeyCode.Unknown` which aren't used by the drivers now, since they return `KeyCode.Null` if KeyChar is `'\0'`. Thus avoids using the prior to check if a key is valid or not and only compare if it's `KeyCode.Null`.
> > 
> > 
> > I agree.
> 
> Note that `Key` already doesn't have `Unknown`.

I think it should also been removed from the `KeyCode` enum.

_Originally posted by @BDisp in https://github.com/gui-cs/Terminal.Gui/issues/2927#issuecomment-1858909499_

### PR DESCRIPTION
**Pull request by tig**
Fixes #3055

KeyCode.ToString now prints "Null" on Null.

Had to fix Menu.

### CODE REVIEW

**Comment by tig** [comment]
@BDisp, this good for you?

**Comment by BDisp** [comment]
Ok I see your are using `Key.Empty` to check if it's a valid rather than `KeyCode.Null`. Good work. Thanks.

**Comment by BDisp** [comment]
> @BDisp, this good for you?

Yes thanks.

File: Terminal.Gui/Views/Menu/Menu.cs [hunk 1]
```diff
@@ -432,7 +432,7 @@ void AddKeyBindings (MenuBarItem menuBarItem)
 		foreach (var menuItem in menuBarItem.Children.Where (m => m != null)) {
 			KeyBindings.Add ((KeyCode)menuItem.HotKey.Value, Command.ToggleExpandCollapse);
 			KeyBindings.Add ((KeyCode)menuItem.HotKey.Value | KeyCode.AltMask, Command.ToggleExpandCollapse);
-			if (menuItem.Shortcut != KeyCode.Unknown) {
+			if (menuItem.Shortcut != KeyCode.Null) {
```
Comment (BDisp) [inline Terminal.Gui/Views/Menu/Menu.cs:435]: Here I've doubt if only a Shift, Alt or Ctrl pass.

Comment (BDisp) [reply Terminal.Gui/Views/Menu/Menu.cs:435]: Ok but `Alt` is acceptable, right?

**Review by BDisp — APPROVED** [review]
It seems all good enough to me. Thanks.

File: Terminal.Gui/Views/Menu/Menu.cs (hunk 1 above)
Comment (tig) [reply Terminal.Gui/Views/Menu/Menu.cs:435]: Regarding a key event with ONLY shift modifiers.... We should ensure they never get sent by drivers.

In otherwords, nobody should REALLY have to call `Key.IsValid`. 

Thus, i think this is ok.

Comment (tig) [reply Terminal.Gui/Views/Menu/Menu.cs:435]: Also, https://github.com/gui-cs/Terminal.Gui/issues/2975 will revamp all of this...

**Pull request merged**